### PR TITLE
Add gradient clip from master

### PR DESCRIPTION
**Pull request by YangFei1990**
create a new PR based on master to resolve conflicts

### CODE REVIEW

File: MaskRCNN/model/generalized_rcnn.py [hunk 1]
```diff
@@ -42,6 +68,8 @@ def optimizer(self):
         opt = tf.train.MomentumOptimizer(lr, 0.9)
         if cfg.TRAIN.NUM_GPUS < 8:
             opt = optimizer.AccumGradOptimizer(opt, 8 // cfg.TRAIN.NUM_GPUS)
+        if cfg.TRAIN.GRADIENT_CLIP:
```
Comment (armandmcqueen) [inline MaskRCNN/model/generalized_rcnn.py]: I prefer to explicitly check, e.g. `if cfg.TRAIN.GRADIENT_CLIP != 0`

File: infra/eks/maskrcnn/templates/maskrcnn.yaml [hunk 2]
```diff
@@ -58,6 +58,7 @@ spec:
         - DATA.BASEDIR=/{{ .Values.maskrcnn.data_fs }}/{{ .Values.maskrcnn.data_dir }}
         - DATA.TRAIN={{ .Values.maskrcnn.data_train }}
         - DATA.VAL={{ .Values.maskrcnn.data_val }}
+        - TRAIN.GRADIENT_CLIP={{ .Values.maskrcnn.gradient_clip }}
```
Comment (armandmcqueen) [inline infra/eks/maskrcnn/templates/maskrcnn.yaml:61]: Please also update `docker/train.sh` and `ami/no_batch_train_1node_16xl_convergence.sh` as well so all training script stay consistent

**Comment by armandmcqueen** [comment]
Looks good. Please address the minor comments and then you can merge once test results show no decline in accuracy.

**Comment by YangFei1990** [comment]
For 32x4, with gradient clip threshold 0.36, the original base learning rate(0.00125) will deliver bad accuracy: 0.36972/0.34077. After change the base lr to 0.0015, the accuracy come back to target: 0.3723/0.34307.
NaN never happened for this branch.
